nixos/modules/services/hardware/xow.nix

```diff
@@ -0,0 +1,29 @@
+{ config, pkgs, lib, ... }:
+
+let
+  cfg = config.services.xow;
+in {
+  options.services.xow = {
+    enable = lib.mkOption {
+      type = lib.types.bool;
+      default = false;
+      description = ''
+        Whether to enable xow or not.
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment.systemPackages = [ pkgs.xow ];
+
+    systemd.services.xow = {
+      wantedBy = [ "multi-user.target" ];
+      description = "Xbox One Wireless Dongle Driver";
+      serviceConfig = {
+        ExecStart = ''
+          ${pkgs.xow}/bin/xow
```

Comment:
This runs the service with the `root` user which should be avoided. The simplest way is to use `serviceConfig.DynamicUser = true;`, though you might need some additional settings to make it work.

Reply:
I naively tried just adding `DynamicUser`, which failed. The program seems to want to `open("/dev/uinput", ...)` which is owned by root. I've contacted the author, see 

https://github.com/medusalix/xow/issues/13

I'm not sure how to fix this myself.

Reply:
Okay, I've rewritten the service to add a udev rule and a new user/group for uinput. It seems to work. Do you think it's okay this way?